### PR TITLE
Initial import of asset build tooling and ui components

### PR DESCRIPTION
**Pull request by bloodearnest**
This attempts to extract the asset build tooling and component library
from job-server, and add it to airlock.  It isolates all this
tooling within an `assets` django app, and exposes a justfile to
interact with it.

You can build the assets with: `just assets/build`. `collectstatic` will
work as expected.

We also add the components browser view at the /ui-components. This acts
as a test to see everything works, and a helpful builtin reference for
using the slippers components.

Some of the components are a job-server specific, e.g. header and
footer, but are preserved as is for now.

We can probably improve on this modularity over time, but this should
allow us to use the UI components as is.

### CODE REVIEW

File: assets/package.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    "htmx.org": "^1.9.10",
+    "just-debounce-it": "^3.2.0",
+    "pretty-bytes": "^6.1.1",
+    "prop-types": "^15.8.1",
+    "simple-datatables": "^9.0.0",
+    "slugify": "^1.6.6",
+    "tailwindcss": "^3.4.1",
+    "wouter": "^2.12.1",
+    "yup": "^1.3.3"
+  }
```
Comment (tomodwyer) [inline assets/package.json:59]:
```suggestion
  "devDependencies": {
    "autoprefixer": "^10.4.16",
    "eslint": "^8.56.0",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-config-prettier": "^9.1.0",
    "eslint-plugin-import": "^2.29.1",
    "eslint-plugin-jsx-a11y": "^6.8.0",
    "eslint-plugin-prettier": "^5.1.3",
    "eslint-plugin-testing-library": "^6.2.0",
    "postcss-import": "^16.0.0",
    "postcss-nesting": "^12.0.2",
    "postcss-preset-env": "^9.3.0",
    "prettier": "^3.2.2",
    "prettier-eslint": "^16.2.0",
    "vite": "^5.0.11",
    "vite-plugin-static-copy": "^1.0.0",
    "vitest": "^1.2.0",
  },
  "dependencies": {
    "@fontsource/public-sans": "^5.0.16",
    "@github/filter-input-element": "^0.1.1",
    "@remotedevforce/tom-select": "^2.2.8",
    "@tailwindcss/forms": "^0.5.7",
    "@tailwindcss/typography": "^0.5.10",
    "@zachleat/details-utils": "^2.0.2",
    "highlight.js": "^11.9.0",
    "html2pdf.js": "^0.10.1",
    "htmx.org": "^1.9.10",
    "simple-datatables": "^9.0.0",
    "slugify": "^1.6.6",
    "tailwindcss": "^3.4.1",
  }
```

File: assets/src/scripts/_plausible.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  if (isLoggedIn) {
+    script.setAttribute("event-is_logged_in", isLoggedIn.content);
+  }
+
+  if (isStaff) {
+    script.setAttribute("event-is_staff", isStaff.content);
+  }
+
+  document.head.appendChild(script);
+}
```
Comment (tomodwyer) [inline assets/src/scripts/_plausible.js:20]: This file can be deleted along with the import

File: assets/src/scripts/base.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+// Web components
+import "@zachleat/details-utils";
+
+// Styles
+import "../styles/base.css";
+
+// Custom JS
+import "./_alert";
+import "./_expand-list";
+import "./_plausible";
```
Comment (tomodwyer) [inline assets/src/scripts/base.js:16]:
```suggestion
```

**Review by evansd — APPROVED** [review]

**Comment by madwort** [comment]
I get some warnings when I try to use `just manage runserver` on a clean checkout:

```
WARNINGS:
?: (django_vite.W001) Cannot read Vite manifest file for app default at /Users/tom/Documents/bennett-institute/airlock2/assets/dist/.vite/manifest.json : [Errno 2] No such file or directory: '/Users/tom/Documents/bennett-institute/airlock2/assets/dist/.vite/manifest.json'
	HINT: Make sure you have generated a manifest file, and that DJANGO_VITE["default"]["manifest_path"] points to the correct location.
?: (staticfiles.W004) The directory '/Users/tom/Documents/bennett-institute/airlock2/assets/dist' in the STATICFILES_DIRS setting does not exist.
```
also the homepage fails with 
```
DjangoViteAssetNotFoundError at /

Cannot find src/scripts/base.js for app=default in Vite manifest at /Users/tom/Documents/bennett-institute/airlock2/assets/dist/.vite/manifest.json
```

**Comment by bloodearnest** [comment]
> I get some warnings when I try to use `just manage runserver` on a clean checkout:
> 
> ```
> WARNINGS:
> ?: (django_vite.W001) Cannot read Vite manifest file for app default at /Users/tom/Documents/bennett-institute/airlock2/assets/dist/.vite/manifest.json : [Errno 2] No such file or directory: '/Users/tom/Documents/bennett-institute/airlock2/assets/dist/.vite/manifest.json'
> 	HINT: Make sure you have generated a manifest file, and that DJANGO_VITE["default"]["manifest_path"] points to the correct location.
> ?: (staticfiles.W004) The directory '/Users/tom/Documents/bennett-institute/airlock2/assets/dist' in the STATICFILES_DIRS setting does not exist.
> ```
> 
> also the homepage fails with
> 
> ```
> DjangoViteAssetNotFoundError at /
> 
> Cannot find src/scripts/base.js for app=default in Vite manifest at /Users/tom/Documents/bennett-institute/airlock2/assets/dist/.vite/manifest.json
> ```

Right, you need to have built assets and collected first.

`just run` ensures that that happens.

However, the latest commit removes the build step, but you'll still need to collectstatic before runserver.

**Pull request merged**